generator/src/test/java/com/scottlogic/deg/generator/cucumber/Random.feature

```diff
@@ -0,0 +1,145 @@
+    And foo contains anything but null
+
+Scenario: The generator produces valid 'String' data in random mode
+  Given foo is of type "string"
+    And foo is anything but null
+    And the generator can generate at most 5 rows
+    And foo is shorter than 10
+  Then 5 rows of data are generated
+    And foo contains string data
+    And foo contains strings of length between 0 and 9 inclusively
+    And foo contains anything but null
+
+Scenario: The generator produces valid RegEx restricted 'String' data in random mode
+  Given foo is of type "string"
+    And foo is anything but null
+    And the generator can generate at most 5 rows
+    And foo is matching regex /[a-z]{0,9}/
+  Then 5 rows of data are generated
+    And foo contains string data
+    And foo contains strings matching /[a-z]{0,9}/
+    And foo contains anything but null
+
+Scenario: The generator produces valid inverted RegEx restricted 'String' data in random mode
+  Given foo is of type "string"
+    And foo is anything but null
+    And the generator can generate at most 5 rows
+    And foo is anything but matching regex /[a-z]{0,9}/
+  Then 5 rows of data are generated
+    And foo contains string data
+    And foo contains anything but strings matching /[a-z]{0,9}/
+    And foo contains anything but null
+
+Scenario: The generator produces valid 'Null' data in random mode
+  Given foo is null
+    And the generator can generate at most 5 rows
+  Then the following data should be generated:
+    | foo  |
+    | null |
+    | null |
+    | null |
+    | null |
+    | null |
+
+  ### Violating ###
+
+Scenario Outline: The generator produces violating (incorrect type) data in random mode for all types
+  Given foo is of type "<type>"
+    And the generator can generate at most 5 rows
+    And the data requested is violating
+  Then 5 rows of data are generated
+    And foo contains anything but <type> data
+
+  Examples:
+    | type    |
+    | string  |
+    | temporal|
+    | numeric |
+
+Scenario: The generator produces violating 'Null' data in random mode
+  Given foo is null
+    And the generator can generate at most 5 rows
+    And the data requested is violating
+  Then 5 rows of data are generated
+    And foo contains anything but null
+
+Scenario: The generator produces violating (not type) 'Temporal' data in random mode
+  Given foo is of type "temporal"
+    And foo is anything but null
+    And the generator can generate at most 5 rows
+    And foo is before or at 2019-01-01T00:00:00.000
+    And the data requested is violating
+    And we do not violate any of type constraints
+  Then 5 rows of data are generated
+    And foo contains temporal data
+    And foo contains temporal values after or at 2019-01-01T00:00:00.001
```

Comment:
With the before test we specified a maximum - is there one for the after test? Also this shouldn't be `after or at` as the at value is valid.